### PR TITLE
[pkg/otlp/model] Send first value for cumulative monotonic sums

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?
Report the first value if the timeseries started after the Datadog Agent process started. Also add a helper function `getProcessStartTime` to retrieve Agent start time.

### Motivation

We currently don’t send the first value of cumulative monotonic sums: we keep it in memory and send the difference with the next value.

On the Agent's OpenMetrics integration, we report the first value if the timeseries started after the Datadog Agent process started.

There is a customer interested in this, they are using OpenMetrics and want to move to OpenTelemetry, and this is a necessary step for them.

### Describe how to test/QA your changes

- [ ] Send an OTLP Counter metric through OTLP ingest in the Agent and sets its start timestamp to be after Agent started. Check that the first count metric is present in-app.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/otlp/model/translator/metrics_translator.go [hunk 1]
```diff
@@ -153,6 +160,11 @@ func (t *Translator) mapNumberMonotonicMetrics(
...
 		}
 
+		if i == 0 && getProcessStartTime() < startTs {
+			// Report the first value if the timeseries started after the Datadog Agent process started.
+			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, val)
+		}
+
 		if dx, ok := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val); ok {
 			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, dx)
 		}
```
Comment (mx-psi) [inline pkg/otlp/model/translator/metrics_translator.go:170]: The way I was thinking about it was this:

```suggestion
		if dx, ok := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val); ok {
			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, dx)
		} else if i == 0 && getProcessStartTime() < startTs {
			// Report the first value if the timeseries started after the Datadog Agent process started.
			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, val)
		}
```

since the first point may have come on a previous payload. In general, I think we should never record `val` if we have something in cache related to a given timeseries.

Does that make sense?

**Review by maycmlee — APPROVED** [review]
👍 for docs

File: pkg/otlp/model/translator/metrics_translator.go [hunk 2]
```diff
@@ -128,14 +129,20 @@ func (t *Translator) mapNumberMetrics(
 	}
 }
 
+// getProcessStartTime returns the start time of the Agent process in seconds since epoch
+func getProcessStartTime() uint64 {
+	return uint64(config.StartTime.Unix())
+}
```
[the remark this answers is not in the document]
Comment (songy23) [reply pkg/otlp/model/translator/metrics_translator.go]: I see, I think that's why the tidy checker complained about `pkg/config` dependency. Starting with a global `startTime` makes sense, I also left a TODO if we ever want to change it to Translator start time.

File: pkg/otlp/model/translator/metrics_translator.go [hunk 3]
```diff
@@ -128,14 +129,20 @@ func (t *Translator) mapNumberMetrics(
...
 // mapNumberMonotonicMetrics maps monotonic datapoints into Datadog metrics
 func (t *Translator) mapNumberMonotonicMetrics(
 	ctx context.Context,
 	consumer TimeSeriesConsumer,
 	dims *Dimensions,
 	slice pmetric.NumberDataPointSlice,
 ) {
-	for i := 0; i < slice.Len(); i++ {
+	i := 0
+	for ; i < slice.Len(); i++ {
```
[the remark this answers is not in the document]
Comment (songy23) [reply pkg/otlp/model/translator/metrics_translator.go]: In my initial version I was using `i` outside the loop, now that's no longer needed. Reverted this change.

File: pkg/otlp/model/translator/metrics_translator.go (hunk 1 above)
Comment (songy23) [reply pkg/otlp/model/translator/metrics_translator.go]: Makes sense to me, this way it also makes the (exclusive) conditions clearer. Updated.

**Review by ahmed-mez — APPROVED** [review]
The changes in `pkg/trace` lgtm.

File: pkg/otlp/model/translator/metrics_translator_test.go [hunk 4]
```diff
@@ -354,6 +354,33 @@ func TestMapIntMonotonicWithReboot(t *testing.T) {
 	)
 }
 
+func TestMapIntMonotonicReportFirstValue(t *testing.T) {
```
Comment (mx-psi) [inline pkg/otlp/model/translator/metrics_translator_test.go:357]: Should we add a test case where the cache already has some data about the timeseries, to test that we send the diff in this situation?

File: pkg/otlp/model/translator/metrics_translator_test.go [hunk 5]
```diff
@@ -354,6 +354,33 @@ func TestMapIntMonotonicWithReboot(t *testing.T) {
...
+		point := slice.AppendEmpty()
+		point.SetStartTimestamp(seconds(startTs))
+		point.SetTimestamp(seconds(startTs + i))
+		point.SetIntValue(val)
+	}
+
+	ctx := context.Background()
+	tr := newTranslator(t, zap.NewNop())
+	consumer := &mockTimeSeriesConsumer{}
+	tr.mapNumberMonotonicMetrics(ctx, consumer, exampleDims, slice)
```
Comment (mx-psi) [inline pkg/otlp/model/translator/metrics_translator_test.go:373]: nit: Despite what other tests do, I think we should write this in terms of `MapMetrics` instead if it's not too much hassle, this will make it easier to refactor code without rewriting tests

You can also do this later when you rewrite the tests in terms of JSON, up to you!

File: pkg/otlp/model/go.mod [hunk 6]
```diff
@@ -20,17 +20,21 @@ require (
 	github.com/dustin/go-humanize v1.0.0 // indirect
```
Comment (mx-psi) [inline pkg/otlp/model/go.mod:20]: We can revert the changes on the `go.mod` and `go.sum` files from both Go modules, they are now not related to this patch and we will avoid introducing unrelated changes on this PR

**Review by mx-psi — APPROVED** [review]

**Comment by songy23** [comment]
Thanks for the review!
